playground/AzureContainerApps/AzureContainerApps.AppHost/infra/main.bicep

```diff
@@ -0,0 +1,76 @@
+targetScope = 'subscription'
+
+@minLength(1)
+@maxLength(64)
+@description('Name of the environment that can be used as part of naming resource convention, the name of the resource group for your application will use this name, prefixed with rg-')
+param environmentName string
+
+@minLength(1)
+@description('The location used for all deployed resources')
+param location string
+
+@description('Id of the user or app to assign application roles')
+param principalId string = ''
+
+@metadata({azd: {
+  type: 'generate'
+  config: {length:22,noSpecial:true}
+  }
+})
+@secure()
+param cache_password string
+param certificateName string
+param customDomain string
+@secure()
+param secretparam string
+
+var tags = {
+  'azd-env-name': environmentName
+}
+
+resource rg 'Microsoft.Resources/resourceGroups@2022-09-01' = {
+  name: 'rg-${environmentName}'
+  location: location
+  tags: tags
+}
+
+module account 'account/account.module.bicep' = {
+  name: 'account'
+  scope: rg
+  params: {
+    keyVaultName: infra.outputs.secret_output_account
+    location: location
+  }
+}
+module api_roles 'api-roles/api-roles.module.bicep' = {
+  name: 'api-roles'
+  scope: rg
+  params: {
+    location: location
+    storage_outputs_name: storage.outputs.name
+  }
+}
+module infra 'infra/infra.module.bicep' = {
+  name: 'infra'
+  scope: rg
+  params: {
+    location: location
+    principalId: ''
```

Comment:
@vhvb1989 This is the last issue with this PR. There's no way to access known parameters from main in the manifest so there's have no way to represent that I need the parameter principlId cleanly.

Reply:
How do you want to express it in the manifest?
 
We can do:
"princiaplId", as raw string, which becomes a reserved word for azd parsing the manifest
 
"{ .princiaplId }" , as a property of a global object.  Manifest in this case knows too much, maybe
 
"{ principalId }" Similar to the raw string, but using the brackets. 
 
"{ _publisher_.principalId }" .  Introduce the concept of a publisher in the manifest. An special way to request functionality for anyone parsing the manifest

Reply:
Agreed on a new known parameter "userPrincipalId"